compiler/rustc_mir_transform/src/ctfe_limit.rs

```diff
@@ -0,0 +1,56 @@
+use crate::MirPass;
+
+use rustc_middle::mir::{
+    BasicBlock, BasicBlockData, BasicBlocks, Body, Statement, StatementKind, TerminatorKind,
+};
+use rustc_middle::ty::TyCtxt;
+
+pub struct CtfeLimit;
```

Comment:
Please add a doc comment explaining what this does. I first thought it only adds `ConstEvalCounter` for back edges, but it also adds one right before every call, it seems?